kitsune/users/views.py

```diff
@@ -44,9 +45,13 @@ def user_auth(request, notification=None):
     """
     next_url = get_next_url(request) or reverse('home')
 
+    aaq = re.match(r"^/[\w\-]+/questions/new/([\w\-]+)/[\w\-]+$", next_url)
+    aaq_product = aaq.group(1) if aaq else None
+
     return render(request, 'users/auth.html', {
         'next_url': next_url,
-        'notification': notification
+        'notification': notification,
+        'aaq_product': aaq_product
```

Comment:
How about removing the regex and adding something like 
```python
{
...
...
'in_aaq_flow': True if request.session['in-aaq'] or False
```
Probably the in_aaq_flow is not a good name but rather an example. Wouldn't that work? It feels to me that's much simpler than the regex and probably more robust since we don't depend on the next_url

Reply:
That cookie won't get set if a user goes directly to the `/questions/new/desktop/form` url (for instance), which we'll eventually be linking to from around the site with the "ask the community" widget we want to add. Also the cookie doesn't tell us what product the user wants to aaq about (which we need to go direct to the form).

The regex should be robust since it's a near-direct copy of what we're using in urls.py

Reply:
The regex might be, the next_url though is not. Ideally the entry point for everything should be only the `aaq` view which will handle all the flow. Having multiple entry points for the aaq is prone to errors

Reply:
But the `aaq` view isn't called, because step 3 is wrapped in the login_required decorator. I don't see why next_url isn't robust? It's the only way users will get back to the aaq flow after login, if it's not set they'll end up on the homepage.